ml/src/main/scala/ai/h2o/sparkling/ml/params/H2OCommonParams.scala

```diff
@@ -21,10 +21,8 @@ import org.apache.spark.ml.param._
 /**
   * This trait contains parameters that are shared across all algorithms.
   */
-trait H2OCommonParams extends H2OMOJOAlgoSharedParams {
+trait H2OCommonParams extends H2OMOJOAlgoSharedParams with H2OAlgoParamsBase {
```

Comment:
nit: I was just thinking about renaming `H2OMOJOAlgoSharedParams` to `H2OBaseMOJOParams`. Then it would be consistent with `H2OSupervisedMOJOParams`, `H2OTreeBasedSupervisedMOJOParams`. 

If we decide to do it, I would also rename `H2OTargetEncoderParams` to `H2OTargetEncoderMOJOParams` so it is all consistent.

All these classes are in scoring package in `ai.h2o.sparkling.params`

Reply:
NP, will do.